File: cpp/reverse-string/reverse_string.cpp
#include "reverse_string.h"
#include <string>

namespace reverse_string {
std::string reverse_string(std::string str) {
  std::string new_string = "";

  for (char c : str) {
    new_string = c + new_string;
  }

  return new_string;
}
} // namespace reverse_string

File: solutions/javascript/reverse-string/2/reverse-string.js
//
// This is only a SKELETON file for the 'Reverse String' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

/**
 * Return a reversed string of `str`
 *
 * @param {string} str
 * @returns {string}
 */
export const reverseString = (str) => {
  let newStr = ''

  for (const c of str) {
    newStr = c + newStr
  }

  return newStr
}
